src/Microsoft.AspNetCore.Authentication/VirtualAuthenticationHandler.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    /// <summary>
+    /// Forwards calls to another authentication scheme.
+    /// </summary>
+    public class VirtualAuthenticationHandler : IAuthenticationHandler, IAuthenticationSignInHandler
+    {
+        protected IOptionsMonitor<VirtualSchemeOptions> OptionsMonitor { get; }
+        public AuthenticationScheme Scheme { get; private set; }
+        public VirtualSchemeOptions Options { get; private set; }
+        protected HttpContext Context { get; private set; }
+
+        public VirtualAuthenticationHandler(IOptionsMonitor<VirtualSchemeOptions> options)
+        {
+            OptionsMonitor = options;
+        }
+
+        /// <summary>
+        /// Initialize the handler, resolve the options and validate them.
+        /// </summary>
+        /// <param name="scheme"></param>
+        /// <param name="context"></param>
+        /// <returns>A Task.</returns>
+        public Task InitializeAsync(AuthenticationScheme scheme, HttpContext context)
+        {
+            if (scheme == null)
+            {
+                throw new ArgumentNullException(nameof(scheme));
+            }
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            Scheme = scheme;
+            Context = context;
+
+            Options = OptionsMonitor.Get(Scheme.Name) ?? new VirtualSchemeOptions();
```

Comment:
You dropped the Validate call that other handlers use?

Reply:
The fallback to a default doesn't make sense in this case, the default is unusable. Throw? Validate would also cover that. It seems like you must set at least one target and/or DefaultTargetSelector, or else what's the point of adding this handler?

Reply:
The default is usable, it just means use the existing defaults for this virtual scheme.  So a use case could be, you want a virtual scheme that only kicks in for some very specific requests, otherwise it falls back to the existing schemes/defaults.  i.e. you have a selector that only targets "/specialmagichacks/" paths

But yeah I guess the Validate on the options is still a useful pattern to have for people who extend the handler/options.  I'll add that back as a no-op

Reply:
The default VirtualSchemeOptions doesn't have a sector either. It's useless without at least one option set.